### PR TITLE
updated api.json

### PR DESCRIPTION
**Pull request by jawspeak**
@tristansokol @matanpaul @alecholmes  @bunnyc1986

### CODE REVIEW

File: api.json [hunk 1]
```diff
@@ -646,6 +691,97 @@
           }
         }
       }
+    },
+    "/v2/locations/{location_id}/orders/{order_id}": {
```
Comment (bunnyc1986) [inline api.json]: RetrieveOrder is not available yet.

File: api.json [hunk 2]
```diff
@@ -22,7 +22,10 @@
         ],
         "summary": "Charge",
         "operationId": "Charge",
-        "description": "Charges a card represented by a token.",
+        "description": "Charges a card represented by a card nonce or a customer\u0027s card on file.\n\nYour request to this endpoint must include _either_:\n\n- A value for the `card_nonce` parameter (to charge a card nonce generated\nwith the `SqPaymentForm`)\n- Values for the `customer_card_id` and `customer_id` parameters (to charge\na customer\u0027s card on file)\n\nIn order for an e-commerce payment to potentially qualify for\n[Square chargeback protection](https://squareup.com/help/article/5394), you\n_must_ provide values for the following parameters in your request:\n\n- `buyer_email_address`\n- At least one of `billing_address` or `shipping_address`\n\nWhen this response is returned, the amount of Square\u0027s processing fee might not yet be\ncalculated. To obtain the processing fee, wait about ten seconds and call\n[RetrieveTransaction](#endpoint-retrievetransaction). See the `processing_fee_money`\nfield of each [Tender included](#type-tender) in the transaction.",
+        "x-oauthpermissions": [
+          "PAYMENTS_WRITE"
```
Comment (tristansokol) [inline api.json]: Why are we using this vs the swagger spec compliant version?

File: api.json [hunk 3]
```diff
@@ -1803,14 +2301,191 @@
...
+      "properties": {
+        "reference_id": {
+          "maxLength": 40,
+          "type": "string",
+          "description": "An optional ID you can associate with the order for your own\npurposes (such as to associate the order with an entity ID in your\nown database).\n\nThis value cannot exceed 40 characters."
+        },
+        "line_items": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LineItem"
```
Comment (tristansokol) [inline api.json]: this isn't defined anywhere, fatal error.

File: api.json [hunk 4]
```diff
@@ -1201,15 +1347,15 @@
         "REJECTED",
         "FAILED"
       ],
-      "description": ""
+      "description": "Indicates a refund\u0027s current status."
```
Comment (tristansokol) [inline api.json]: All of these descriptions are not swagger compliant, you can see all the warnings if you try pasting into the swagger editor.

File: api.json [hunk 5]
```diff
@@ -1803,14 +2301,191 @@
...
+        }
+      },
+      "x-sq-sdk-sample-code": {
+        "python": "/sdk_samples/CreateCustomer/CreateCustomerResponse.python",
+        "csharp": "/sdk_samples/CreateCustomer/CreateCustomerResponse.csharp",
+        "php": "/sdk_samples/CreateCustomer/CreateCustomerResponse.php",
+        "ruby": "/sdk_samples/CreateCustomer/CreateCustomerResponse.ruby"
+      }
+    },
+    "CreateOrderRequest": {
```
Comment (tristansokol) [inline api.json]: This definition isn't used.

File: api.json [hunk 6]
```diff
@@ -1803,14 +2301,191 @@
...
+                  "amount": 2000,
+                  "currency": "USD"
+                }
+              }
+            ]
+          }
+        }
+      }
+    },
+    "CreateOrderRequestLineItem": {
```
Comment (tristansokol) [inline api.json]: This definition isn't used.

File: api.json [hunk 7]
```diff
@@ -1803,14 +2301,191 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LineItem"
+          },
+          "description": "The line items to associate with this order.\n\nEach line item represents a different product (or a custom monetary amount)\nto include in a purchase."
+        }
+      },
+      "description": "The object describes the order."
+    },
+    "CreateOrderResponse": {
```
Comment (tristansokol) [inline api.json]: this definition isn't used.

File: api.json (hunk 1 above)
Comment (tristansokol) [inline api.json]: the 🐈  is out of the 👜 now.

File: api.json (hunk 3 above)
Comment (jawspeak) [inline api.json]: fixed

File: api.json (hunk 2 above)
Comment (jawspeak) [inline api.json]: discussed offline

**Comment by matanpaul** [comment]
lgtm

**Pull request merged**